src/mimecodec.ts

```diff
@@ -396,70 +427,93 @@ export function parseHeaderValue (str) {
   }
 
   if (type === 'value') {
-    if (key === false) {
-      response.value = value.trim()
+    if (!key) {
+      responseValue = value.trim()
     } else {
-      response.params[key] = value.trim()
+      initialParams[key] = value.trim()
     }
   } else if (value.trim()) {
-    response.params[value.trim().toLowerCase()] = ''
+    initialParams[value.trim().toLowerCase()] = ''
   }
 
   // handle parameter value continuations
   // https://tools.ietf.org/html/rfc2231#section-3
 
+  const processedParams: Record<
```

Comment:
This part had the most actual code changes. Instead of working in the same object, I've split each step to create a new object, so that it can be statically analyzed by TS